### PR TITLE
Enable blocking RPC methods

### LINKED ISSUE #1018: Resolve long running RPC calls halting node sync
It was noted that calls such as:
- `eth_call`
- `eth_estimateGas`

among others cause the node's sync to halt / stall. 

Investigate & resolve

### PR DESCRIPTION
**Pull request by rakanalh**
# Description

## Linked Issues
- Fixes #1018

### CODE REVIEW

File: crates/evm/src/query.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ pub struct EstimatedDiffSize {
 #[rpc_gen(client, server)]
 impl<C: sov_modules_api::Context> Evm<C> {
     /// Handler for `net_version`
-    #[rpc_method(name = "net_version")]
+    #[rpc_method(name = "net_version", blocking)]
```
Comment (rakanalh) [inline crates/evm/src/query.rs:93]: This marker basically tells jsonrpsee that we want to `spawn_blocking` this call and await it internally.

Comment (kpp) [inline crates/evm/src/query.rs:93]: fix

File: crates/evm/src/query.rs [hunk 2]
```diff
@@ -106,7 +106,7 @@ impl<C: sov_modules_api::Context> Evm<C> {
 
     /// Handler for: `eth_chainId`
     #[rpc_method(name = "eth_chainId")]
-    pub fn chain_id(&self, working_set: &mut WorkingSet<C>) -> RpcResult<Option<U64>> {
+    pub async fn chain_id(&self, working_set: &mut WorkingSet<C>) -> RpcResult<Option<U64>> {
```
Comment (kpp) [inline crates/evm/src/query.rs:109]: fix

File: crates/sovereign-sdk/module-system/sov-modules-macros/src/rpc/rpc_gen.rs [hunk 3]
```diff
@@ -171,6 +208,7 @@ impl RpcImplBlock {
             quote! {
                 /// Allows a Runtime to be converted into a functional RPC server by simply implementing the two required methods -
                 /// `get_backing_impl(&self) -> MyModule` and `get_working_set(&self) -> ::sov_modules_api::WorkingSet<C>`
+                #[async_trait::async_trait]
```
Comment (kpp) [inline crates/sovereign-sdk/module-system/sov-modules-macros/src/rpc/rpc_gen.rs:211]: Add only if > 0 method.method_signature.asyncness

**Review by kpp — APPROVED** [review]

**Comment by kpp** [comment]
The thing is I doubt we will ever have async endpoints. We would have need them for things like going to network to get some info, but due do the nature of our architecture that's not the case...

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**